src/joystream-lib/api.ts

```diff
@@ -212,6 +228,122 @@ export class JoystreamJs {
     })
   }
 
+  private async _createOrUpdateVideo(
+    updatedVideoId: VideoId | null = null,
+    memberId: MemberId,
+    channelId: ChannelId,
+    inputMetadata: CreateVideoMetadata,
+    inputAssets: VideoAssets,
+    cb?: ExtrinsicStatusCallbackFn
+  ): Promise<ExtrinsicResult<VideoId>> {
+    const newVideo = updatedVideoId === null
+
+    // === video assets ===
+    // first video, then thumbnail
+    const newAssetsList: NewAsset[] = [
+      ...(inputAssets.video ? [inputAssets.video] : []),
+      ...(inputAssets.thumbnail ? [inputAssets.thumbnail] : []),
+    ]
+
+    // === video assets ===
+    const protoMeta = new VideoMetadata()
+    if (inputMetadata.title != null) {
+      protoMeta.setTitle(inputMetadata.title)
+    }
+    if (inputMetadata.description != null) {
+      protoMeta.setDescription(inputMetadata.description)
+    }
+    if (inputMetadata.isPublic != null) {
+      protoMeta.setIsPublic(inputMetadata.isPublic)
+    }
+    if (inputMetadata.language != null) {
+      protoMeta.setLanguage(inputMetadata.language)
+    }
+    if (inputMetadata.isExplicit != null) {
+      protoMeta.setIsExplicit(inputMetadata.isExplicit)
+    }
+    if (inputMetadata.category != null) {
+      protoMeta.setCategory(inputMetadata.category)
+    }
+    if (inputMetadata.duration != null) {
+      protoMeta.setDuration(inputMetadata.duration)
+    }
+    if (inputMetadata.mediaPixelHeight != null) {
+      protoMeta.setMediaPixelHeight(inputMetadata.mediaPixelHeight)
+    }
+    if (inputMetadata.mediaPixelWidth != null) {
+      protoMeta.setMediaPixelWidth(inputMetadata.mediaPixelWidth)
+    }
+    if (inputMetadata.hasMarketing != null) {
+      protoMeta.setHasMarketing(inputMetadata.hasMarketing)
+    }
+
+    if (inputMetadata.mediaType != null) {
+      const protoMediaType = new MediaType()
+      protoMediaType.setCodecName(inputMetadata.mediaType.codecName || '')
+      protoMediaType.setContainer(inputMetadata.mediaType.container || '')
+      protoMediaType.setMimeMediaType(inputMetadata.mediaType.mimeMediaType || '')
```

Comment:
Your create `protoMediaType` instance but you don't use it at all? Unless you do `protoMeta.setMediaType(protoMediaType)` this won't have any effect

Reply:
Same for license and publishedBeforeJoystream

Reply:
🤦🏻  right! Now everything works as it should 😅